api/routes/auth.js

```diff
@@ -0,0 +1,136 @@
+const router = require("express").Router();
+const jsonwebtoken = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
+const User = require("../models/user");
+const { SECRET_PW } = process.env;
+
+router.post("/signup", async (req, res, next) => {
+  const status = 201;
+  try {
+    const { username, password, admin } = req.body;
+    const user = await User.findOne({ username });
+    if (user) throw new Error(`User ${username} already exists.`);
+
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
+    const response = await User.create({
+      username,
+      password: hashedPassword,
+      admin
+    });
+    res.status(status).json({ status, response });
+  } catch (error) {
+    console.log(error);
+    const e = new Error(`User ${username} already exists.`);
+    e.status = 400;
+    next(e);
+  }
+});
+
+router.post("/login", async (req, res, next) => {
+  const status = 201;
+  try {
+    const { username, password } = req.body;
+    const checkForUser = await User.findOne({
+      username
+    });
+    if (!checkForUser) throw new Error(`User ${username} does not exist.`);
+
+    const user = await bcrypt.compare(password, checkForUser.password);
+    if (!user) throw new Error(`Password is invalid.`);
```

Comment:
You'll want to keep these error messages the same, otherwise it's easier for someone to "guess" at a username. Having the same error message is more secure.